### PR TITLE
Bug Fix: Client Login Validation

### PR DESCRIPTION
**Pull request by yaimgod**

### CODE REVIEW

File: source/handlers/server.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def _run(self):
             except socket.timeout:
                 continue
             except Exception as e:
-                print '[-] unknown exception:', e
+                print '[-] unknown exception: ', e
```
Comment (Nisitay) [inline source/handlers/server.py:44]: The space is not necessary, the comma adds a space automatically

File: source/games/game.py [hunk 2]
```diff
@@ -59,3 +59,6 @@ def player_joined(self, player_name):
     def broadcast(self, message):
         for sock in self.sockets:
             sock.send(message.serialize())
+
+    def does_player_exiset(self,sock):
```
Comment (Nisitay) [inline source/games/game.py:63]: The function name has a typo. I think you meant "does_player_exist" but I think "does_sock_exist" would be more fitting

File: source/controllers/game.py [hunk 3]
```diff
@@ -30,13 +30,16 @@ def join_game_controller(args, sock):
...
     player_name = str(args['player_name']).strip()
     password = str(args['password']).strip()
-
-    try:
-        join_game(player_name, game_id, password, sock)
-    except TakiException as e:
-        return e.response()
-    except Exception:
-        return Responses.INTERNAL_ERROR
+    if does_player_exiset(sock):
```
Comment (Nisitay) [inline source/controllers/game.py:33]: Shouldn't the game object check if the socket already exists? (in the "add_player" func in game.py)

**Review by Nisitay — CHANGES_REQUESTED** [review]

**Review by omerk2511 — CHANGES_REQUESTED** [review]
Good job. I would like you to save a list of clients instead of socks in the Game object, and also change the name of the argument in games/games.py and also in the relevant decorators (authenticated and validator).

File: source/controllers/game.py [hunk 4]
```diff
@@ -46,7 +50,7 @@ def join_game_controller(args, sock):
 
 @controller(Code.LEAVE_GAME)
 @authenticated
-def leave_game_controller(user, args, sock):
+def leave_game_controller( client):
```
Comment (omerk2511) [inline source/controllers/game.py:53]: The arguments of this function should be: `user, args, client`

File: source/controllers/game.py [hunk 5]
```diff
@@ -26,17 +27,20 @@ def create_game_controller(args, sock):
...
     player_name = str(args['player_name']).strip()
     password = str(args['password']).strip()
-
-    try:
-        join_game(player_name, game_id, password, sock)
-    except TakiException as e:
-        return e.response()
-    except Exception:
-        return Responses.INTERNAL_ERROR
+    if not client.in_game:
```
Comment (omerk2511) [inline source/controllers/game.py:34]: Please create a `@not_in_game` decorator.

**Review by omerk2511 — CHANGES_REQUESTED** [review]
Excellent work

**Review by omerk2511 — APPROVED** [review]

**Review by Nisitay — APPROVED** [review]

**Pull request merged**